addons/account/models/account_move.py

```diff
@@ -779,26 +779,34 @@ def _compute_highest_name(self):
         for record in self:
             record.highest_name = record._get_last_sequence()
 
+    @api.model
+    def _get_query_made_hole(self, ids=None):
+        return f"""
+                SELECT this.id
+                  FROM account_move this
+                  JOIN res_company company ON company.id = this.company_id
+             LEFT JOIN account_move other ON this.journal_id = other.journal_id
+                                         AND this.sequence_prefix = other.sequence_prefix
+                                         AND this.sequence_number = other.sequence_number + 1
+                 WHERE other.id IS NULL
+                   AND this.sequence_number != 1
+                   AND this.name != '/'
+                  {"AND this.id = ANY(%(move_ids)s)" if ids is not None else ""}
+        """, {'move_ids': ids} if ids is not None else {}
+
     @api.depends('name', 'journal_id')
     def _compute_made_sequence_hole(self):
-        self.env.cr.execute("""
-            SELECT this.id
-              FROM account_move this
-              JOIN res_company company ON company.id = this.company_id
-         LEFT JOIN account_move other ON this.journal_id = other.journal_id
-                                     AND this.sequence_prefix = other.sequence_prefix
-                                     AND this.sequence_number = other.sequence_number + 1
-             WHERE other.id IS NULL
-               AND this.sequence_number != 1
-               AND this.name != '/'
-               AND this.id = ANY(%(move_ids)s)
-        """, {
-            'move_ids': self.ids,
-        })
+        query, params = self._get_query_made_hole(self.ids)
+        self.env.cr.execute(query, params)
         made_sequence_hole = set(r[0] for r in self.env.cr.fetchall())
         for move in self:
             move.made_sequence_hole = move.id in made_sequence_hole
 
+    def _search_made_sequence_hole(self, operator, value):
```

Comment:
```suggestion
    def _search_made_sequence_hole(self, operator, value):
        assert operator in ('=', '!=')
```

Just to avoid people using `in` and `not in` for instance

Reply:
I know it's in the UI, but is It really necessary?
![image](https://github.com/odoo/odoo/assets/6582885/3bcd1ad3-e4b6-47da-879f-4f5fd27b3286)